app/Http/Controllers/User/UserController.php

```diff
@@ -275,8 +278,10 @@ public function createNewDataset(Request $request)
     public function userEditDataset(Request $request, int $id)
     {
         $user = auth()->user();
-        $dataset = $user->datasets()->find($id);
-
+        $dataset = $user->datasets()->with(['dataitems' => function ($query) {
+            $query->orderBy('id');
+        }])->find($id);
+    

```

Comment:
This change is unnecessary. When editing a dataset, the dataitems are not used in this case.